sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala

```diff
@@ -406,56 +411,81 @@ case class GpuSampleExec(lowerBound: Double, upperBound: Double, withReplacement
     val numOutputRows = gpuLongMetric(NUM_OUTPUT_ROWS)
     val numOutputBatches = gpuLongMetric(NUM_OUTPUT_BATCHES)
     val opTime = gpuLongMetric(OP_TIME)
+
     val rdd = child.executeColumnar()
-    if (withReplacement) {
-      new GpuPartitionwiseSampledRDD(
-        rdd,
-        new GpuPoissonSampler(upperBound - lowerBound, useGapSamplingIfPossible = false,
-          numOutputRows, numOutputBatches, opTime),
-        preservesPartitioning = true,
-        seed)
-    } else {
+    val useGpuToSample = new RapidsConf(conf).isFastSampleEnabled
+    if (useGpuToSample) {
+      // CPU inconsistent, use GPU sample
       rdd.mapPartitionsWithIndex(
-        (index, iterator) => {
-          // use CPU sampler generate filter
-          val sampler = new BernoulliCellSampler(lowerBound, upperBound)
-          sampler.setSeed(seed + index)
-          iterator.map[ColumnarBatch] { batch =>
-            numOutputBatches += 1
-            withResource(batch) { b => // will generate new columnar column, close this
-              val numRows = b.numRows()
-              val filter = withResource(HostColumnVector.builder(DType.BOOL8, numRows)) {
-                builder =>
-                  (0 until numRows).foreach { _ =>
-                    val n = sampler.sample()
-                    if (n > 0) {
-                      builder.append(1.toByte)
-                      numOutputRows += 1
-                    } else {
-                      builder.append(0.toByte)
+        (_, iterator) => {
+          iterator.map[ColumnarBatch] { columnarBatch =>
+            withResource(new NvtxWithMetrics("Sample Exec", NvtxColor.YELLOW, opTime)) { _ =>
+              withResource(columnarBatch) { cb =>
+                numOutputBatches += 1
+                val numSampleRows = (cb.numRows() * (upperBound - lowerBound)).toLong
+
+                val colTypes = GpuColumnVector.extractTypes(cb)
+                if (numSampleRows == 0L) {
+                  GpuColumnVector.emptyBatchFromTypes(colTypes)
+                } else if(cb.numCols() == 0) {
+                    // for count agg, num of cols is 0
+                    val c = GpuColumnVector.emptyBatchFromTypes(colTypes)
+                    c.setNumRows(numSampleRows.toInt)
+                    c
+                } else {
+                  withResource(GpuColumnVector.from(cb)) { table =>
+                    withResource(table.sample(numSampleRows, withReplacement, seed)) { sampled =>
```

Comment:
nit: Because we use the same seed every time then I think all of the batches will be sampled the same way. I am not sure mathematically how that all works out.  Could we at least set the seed to `seed + index` so each task does it slightly differently?

Reply:
Updated, use the way like CPU to set the seed.